testutil/capturelog.go

```diff
@@ -4,79 +4,129 @@ import (
 	"fmt"
 	"log"
 	"sync"
+	"testing"
 
 	"github.com/influxdata/telegraf"
 )
 
 var _ telegraf.Logger = &CaptureLogger{}
 
+const (
+	logLevelDebug = 'D'
+	logLevelWarn  = 'W'
+	logLevelInfo  = 'I'
+	logLevelError = 'E'
+)
+
+type logMessage struct {
+	level   byte
+	message string
+}
+
 // CaptureLogger defines a logging structure for plugins.
 type CaptureLogger struct {
-	Name   string // Name is the plugin name, will be printed in the `[]`.
-	errors []string
+	T        *testing.T
+	Name     string // Name is the plugin name, will be printed in the `[]`.
+	messages []logMessage
 	sync.Mutex
 }
 
-// Errorf logs an error message, patterned after log.Printf.
-func (l *CaptureLogger) Errorf(format string, args ...interface{}) {
-	s := fmt.Sprintf("E! ["+l.Name+"] "+format, args...)
+func (l *CaptureLogger) msgString(msg logMessage) string {
+	return fmt.Sprintf("%c! [%s] %s", msg.level, l.Name, msg.message)
+}
+func (l *CaptureLogger) logMsg(msg logMessage) {
 	l.Lock()
-	l.errors = append(l.errors, s)
+	l.messages = append(l.messages, msg)
+	l.T.Log(l.msgString(msg))
```

Comment:
Hmmm I would rather not log the message here. In most of the tests the message is not interesting to the user but only to the test. You can still print the messages afterwards, now they are collected, if you need to in your test.
Removing this will also remove the need to associate the logger and shrink the PR...